dlrover/python/common/multi_process.py

```diff
@@ -267,45 +275,81 @@ class SharedLock(LocalSocketComm):
             the lock.
     """
 
-    def __init__(self, name="", create=False):
-        super().__init__(name, create)
+    def __init__(self, name="", create=False, owner=""):
+        super().__init__(name, create, persist=True)
         if self._create:
             self._lock = threading.Lock()
         else:
             self._lock = None
+        self._id = owner
 
     def _deal_shared_lock_msg(self, connection):
-        try:
-            recv_data = _socket_recv(connection)
-            msg: SocketRequest = pickle.loads(recv_data)
-            if msg.method == "acquire":
-                response = LockAcquireResponse()
-                response.acquired = self.acquire(**msg.args)
-            elif msg.method == "locked":
-                response = LockedResponse()
-                response.locked = self.locked()
-            elif msg.method == "release":
-                self.release()
-            response.status = SUCCESS_CODE
-        except Exception:
-            response = SocketResponse()
-            response.status = ERROR_CODE
-        send_data = pickle.dumps(response)
-        _socket_send(connection, send_data)
+        lock_acquired = False
+
+        while True:
+            try:
+                recv_data = _socket_recv(connection)
+                msg: SocketRequest = pickle.loads(recv_data)
+                if msg.method == "acquire":
+                    response = LockAcquireResponse()
+                    response.acquired = self.acquire(**msg.args)
+                    lock_acquired = response.acquired
+                elif msg.method == "locked":
+                    response = LockedResponse()
+                    response.locked = self.locked()
+                elif msg.method == "release":
+                    self.release()
+                    lock_acquired = False
+                response.status = SUCCESS_CODE
+            except (
+                ConnectionResetError,
+                ConnectionAbortedError,
+                BrokenPipeError,
+                ConnectionError,
+            ) as e:
+                logger.error(f"SharedLock connection error: {e}")
+                break
+            except OSError as e:
+                logger.error(f"SharedLock os error: {e}")
+                break
+            except EOFError as e:
+                logger.error(f"SharedLock eof error: {e}")
+                break
+            except Exception as e:
+                logger.error(f"SharedLock unexpected recv error: {e}")
+                response = SocketResponse()
+                response.status = ERROR_CODE
+
+            try:
+                send_data = pickle.dumps(response)
+                _socket_send(connection, send_data)
+            except Exception as e:
```

Comment:
Should we catch the exception here? The upstream caller won't know that an error occurred here if so.